modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -114,26 +393,232 @@ sealed class IO<out A> : IOOf<A> {
     /* For parMap, look into IOParallel */
   }
 
+  /**
+   * Transform the [IO] wrapped value of [A] into [B] preserving the [IO] structure.
+   *
+   * @param f a pure function that maps the value [A] to a value [B].
+   * @returns an [IO] that results in a value [B].
+   *
+   * ```kotlin:ank:playground
+   * import arrow.effects.IO
+   *
+   * fun main(args: Array<String>) {
+   *   val result =
+   *   //sampleStart
+   *   IO.just("Hello").map { "$it World" }
+   *   //sampleEnd
+   *   println(result.unsafeRunSync())
+   * }
+   * ```
+   */
   open fun <B> map(f: (A) -> B): IO<B> =
     Map(this, f, 0)
 
+  /**
+   * Transform the [IO] value of [A] by sequencing an effect [IO] that results in [B].
+   *
+   * @param f function that returns the [IO] effect resulting in [B] based on the input [A].
+   * @returns an effect that results in [B].
+   *
+   * ```kotlin:ank:playground
+   * import arrow.effects.IO
+   *
+   * fun main(args: Array<String>) {
+   *   val result =
+   *   //sampleStart
+   *   IO.just("Hello").flatMap { IO { "$it World" } }
+   *   //sampleEnd
+   *   println(result.unsafeRunSync())
+   * }
+   * ```
+   */
   open fun <B> flatMap(f: (A) -> IOOf<B>): IO<B> =
     Bind(this) { f(it).fix() }
 
+  /**
+   * Continue the evaluation on provided [CoroutineContext]
+   *
+   * @param ctx [CoroutineContext] to run evaluation on
+   * @returns an [IO] that'll run the following computations on [ctx].
+   *
+   * ```kotlin:ank:playground
+   * import arrow.effects.IO
+   * import kotlinx.coroutines.Dispatchers
+   *
+   * fun main(args: Array<String>) {
+   *   //sampleStart
+   *   val result = IO.unit.continueOn(Dispatchers.Default).flatMap {
```

Comment:
Maybe map is easier for the example?

Reply:
I did this so they could see `Dispatcher-default-work..` in the playgrounds, but maybe better to optimize for inline reading instead of online docs?

Should we do same for methods like `bracketCase` and add the big examples on the typeclass instead? Or even Bracket's file kdoc?

cc\ @raulraja